### ISSUE TITLE
net: tests doesn't pass on IPv6 only machines

### ISSUE THREAD

**Issue by azhavnerchik** [golang/go#9890]
I am trying to build go on IPv6 only machine and seeing following error:
--- FAIL: TestDialTimeout (0.01s)
        dial_test.go:108: got error "dial tcp 127.0.0.1:61521: connection reset by peer"; not a timeout
--- FAIL: TestIPv4MulticastListener (0.00s)
        multicast_test.go:55: First ListenMulticastUDP on <nil> failed: listen udp 224.0.0.254: setsockopt: no such device
FAIL
FAIL    net     2.086s

Test should check if IPv4/IPv6 addresses assigned on the machine before starting test.

**Comment by mikioh** [comment]
This is a duplicate of #5707.
